src/rewrite/mod.rs

```diff
@@ -49,10 +49,14 @@ fn any_to_t<T: Any>(t: Box<dyn Any>) -> T {
 pub fn apply_rewrites<L: Language, N: Analysis<L>>(eg: &mut EGraph<L, N>, rewrites: &[Rewrite<L, N>]) -> bool {
     let prog = eg.progress();
 
+    let search_span = tracing::trace_span!("search").entered();
     let ts: Vec<Box<dyn Any>> = rewrites.iter().map(|rw| (*rw.searcher)(eg)).collect();
+    search_span.exit();
+    let apply_span = tracing::trace_span!("apply").entered();
```

Comment:
are you sure this compiles, if tracing is off?

Reply:
it doesn't, hence my comment above

Reply:
ah I see, there's `#[(cfg(feature = "trace")]` for that.
You can see how it works, as it's everywhere in the explanations code too

Reply:
>  It's kind of anoying to feature gate tracing ourselves in general, knowing that the tracing crate has built-in features to make tracing a no-op ...

ah I see. If that's so, we can also get rid of the feature gate